### DISCUSSION TITLE
Dockerhub latest tag out of date

### DISCUSSION THREAD

**Discussion by netguino** [photoprism/photoprism#4171] [General]
Hello!

Is the dockerhub tag still being updated? The last push is from [4 months ago](https://hub.docker.com/layers/photoprism/photoprism/latest/images/sha256-cbaf35b067efc179dee68ab703db8911c2ad306830831458abb369542ee4cf12?context=explore)  , and there's been many changes since then. 

Is the CI pipeline broken or are changes to dockerhub only pushed on new releases?

If this is the case, how risky is it to run preview? Can this lead to corruption?

From a subscriber, thanks for such a great product.

**Comment by lastzero** [comment]
We are currently preparing our next release, featuring support for 2-Factor Authentication (2FA) and many other improvements. See our repository and GitHub Issues for updates and ongoing activities, for example:

- https://github.com/photoprism/photoprism/issues/808#issuecomment-2046599624

Since any security-related functionality needs to be tested very well and implementing these changes was a more complex undertaking than, say, supporting a new file format or implementing purely visual UI improvements, this is one of the updates that was longer in the making. 

We also updated the reverse geocoding data and default map tiles for you earlier this year (something many users had asked for, but which didn't require updating the Docker image):

- https://github.com/photoprism/photoprism/issues/3849

That being said, you are welcome to help us test the preview image (it most likely won't break anything, but if you want to be on the safe side and don't have time to test, it might be better to wait a few more days until we have tested everything):

- https://docs.photoprism.app/release-notes/#development-preview

Thank you for your support! :)

**Reply by netguino** [reply]
Thanks for the prompt response Michael!

So, just to be clear ( and for future people searching ) is it correct to say that the `latest` tag does get updated just on releases? Makes sense. 

I wonder if there might be benefit to having a `master` tag  as well.

**Reply by lastzero** [reply]
There has always been confusion around the use of the `latest` tag as people assume all sorts of things based on its name. It's the default tag for Docker, used when you just specify an image name like `photoprism/photoprism`.

Since we don't want all our users to use the unstable version (which sometimes gets multiple updates per day!), we use the `preview` tag and sometimes the `test` tag (if we prefer to test with fewer users or internally).

Note that we don't have a master branch on GitHub either, just `develop`, `preview` and `release` - where the Docker tag `preview` corresponds to the code in the `preview` branch, while `test` is usually created from `develop` (but not very often).

Our stable releases have individual Docker tags (based on the date of release) and `latest` then points to the most recent one by convention.

**Reply by netguino** [reply]
Excellent answer. Shall be useful for people in the future. 

Thanks!

**Reply by lastzero** [reply]
Note that an [updated preview build](https://docs.photoprism.app/getting-started/updates/#development-preview) is now available [on Docker Hub](https://hub.docker.com/r/photoprism/photoprism/tags?name=preview) for final testing:

- https://docs.photoprism.app/getting-started/updates/#development-preview
- https://docs.photoprism.app/release-notes/#development-preview

**Any help with that is much appreciated!** :tada:
